ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -192,31 +205,47 @@ def get_pcf_channel_metrics(self, queue_manager, tags, config):
             self.gauge(mname, channels, tags=tags)
 
         # grab all the discoverable channels
-        self._get_channel_status(queue_manager, '*', tags, config)
+        self._submit_channel_status(queue_manager, '*', tags, config)
 
         # check specific channels as well
         # if a channel is not listed in the above one, a user may want to check it specifically,
         # in this case it'll fail
         for channel in config.channels:
-            self._get_channel_status(queue_manager, channel, tags, config)
+            self._submit_channel_status(queue_manager, channel, tags, config)
 
-    def _get_channel_status(self, queue_manager, channel, tags, config):
-        channel_tags = tags + ["channel:{}".format(channel)]
+    def _submit_channel_status(self, queue_manager, search_channel_name, tags, config):
+        """Submit channel status
+        :param search_channel_name might contain wildcard characters
+        """
         try:
-            args = {pymqi.CMQCFC.MQCACH_CHANNEL_NAME: ensure_bytes(channel)}
+            args = {pymqi.CMQCFC.MQCACH_CHANNEL_NAME: ensure_bytes(search_channel_name)}
             pcf = pymqi.PCFExecute(queue_manager)
             response = pcf.MQCMD_INQUIRE_CHANNEL_STATUS(args)
         except pymqi.MQMIError as e:
             self.log.warning("Error getting CHANNEL stats {}".format(e))
+            channel_tags = tags + ["channel:{}".format(search_channel_name)]
             self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.CRITICAL, channel_tags)
         else:
             for channel_info in response:
-                name = channel_info[pymqi.CMQCFC.MQCACH_CHANNEL_NAME]
-                name = name.strip()
+                channel_name = channel_info[pymqi.CMQCFC.MQCACH_CHANNEL_NAME].decode().strip()
+                channel_tags = tags + ["channel:{}".format(channel_name)]
 
-                # running = 3, stopped = 4
-                status = channel_info[pymqi.CMQCFC.MQIACH_CHANNEL_STATUS]
-                if status == 3:
+                channel_status = channel_info[pymqi.CMQCFC.MQIACH_CHANNEL_STATUS]
+
+                if channel_status == pymqi.CMQCFC.MQCHS_RUNNING:
                     self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.OK, channel_tags)
-                elif status == 4:
+                elif channel_status == pymqi.CMQCFC.MQCHS_STOPPING:
                     self.service_check(self.CHANNEL_SERVICE_CHECK, AgentCheck.WARNING, channel_tags)
+
+                self._submit_channel_count(channel_status, channel_tags)
+
+    def _submit_channel_count(self, channel_status, channel_tags):
+        metric_name = 'ibm_mq.channel.count'
```

Comment:
you can make this a constant at the top of the file

Reply:
fixed